src/graphql/node/input.rs

```diff
@@ -58,346 +57,162 @@ pub(super) struct NodeInput {
+            txt: input.txt,
+            smtp_eml: input.smtp_eml,
+            ftp: input.ftp,
+            giganto: input.giganto,
+            giganto_ingestion_ip: parse_str_to_ip(input.giganto_ingestion_ip.as_deref()),
+            giganto_ingestion_port: input.giganto_ingestion_port,
+            giganto_publish_ip: parse_str_to_ip(input.giganto_publish_ip.as_deref()),
+            giganto_publish_port: input.giganto_publish_port,
+            giganto_graphql_ip: parse_str_to_ip(input.giganto_graphql_ip.as_deref()),
+            giganto_graphql_port: input.giganto_graphql_port,
+            retention_period: input.retention_period,
+            reconverge: input.reconverge,
+            reconverge_review_ip: parse_str_to_ip(input.reconverge_review_ip.as_deref()),
+            reconverge_review_port: input.reconverge_review_port,
+            reconverge_giganto_ip: parse_str_to_ip(input.reconverge_giganto_ip.as_deref()),
+            reconverge_giganto_port: input.reconverge_giganto_port,
+            hog: input.hog,
+            hog_review_ip: parse_str_to_ip(input.hog_review_ip.as_deref()),
+            hog_review_port: input.hog_review_port,
+            hog_giganto_ip: parse_str_to_ip(input.hog_giganto_ip.as_deref()),
+            hog_giganto_port: input.hog_giganto_port,
+            protocols: input.protocols,
+            protocol_list: input.protocol_list.clone(),
+            sensors: input.sensors,
+            sensor_list: input.sensor_list.clone(),
+        })
+    }
+}
 
-        // review
-        value.review = self.review;
-        value.review_port = self.review_port;
-        value.review_web_port = self.review_web_port;
+fn parse_str_to_ip(ip_str: Option<&str>) -> Option<IpAddr> {
+    ip_str.and_then(|ip_str| ip_str.parse::<IpAddr>().ok())
+}
 
-        // piglet
-        value.piglet = self.piglet;
-        value.piglet_giganto_ip = if let Some(ip) = self.piglet_giganto_ip.as_deref() {
-            Some(ip.parse::<IpAddr>().context("invalid IP address")?)
-        } else {
-            None
-        };
-        value.piglet_giganto_port = self.piglet_giganto_port;
-        value.piglet_review_ip = if let Some(ip) = self.piglet_review_ip.as_deref() {
-            Some(ip.parse::<IpAddr>().context("invalid IP address")?)
-        } else {
-            None
-        };
-        value.piglet_review_port = self.piglet_review_port;
-        value.save_packets = self.save_packets;
-        value.http = self.http;
-        value.office = self.office;
-        value.exe = self.exe;
-        value.pdf = self.pdf;
-        value.html = self.html;
-        value.txt = self.txt;
-        value.smtp_eml = self.smtp_eml;
-        value.ftp = self.ftp;
+#[allow(clippy::module_name_repetitions)]
+#[derive(Clone, InputObject)]
+pub(super) struct NodeInput {
+    pub name: String,
+    pub name_draft: Option<String>,
+    pub setting: Option<NodeSettingInput>,
+    pub setting_draft: Option<NodeSettingInput>,
```

Comment:
Given `NodeInput` deals with users' input and users cannot directly change `name` and `setting` unless they just click the apply button, why are both present here?

Reply:
done